types/block.go

```diff
@@ -434,65 +436,130 @@ func (h *Header) StringIndented(indent string) string {
 
 //-------------------------------------
 
-// CommitSig is a vote included in a Commit.
-// For now, it is identical to a vote,
-// but in the future it will contain fewer fields
-// to eliminate the redundancy in commits.
-// See https://github.com/tendermint/tendermint/issues/1648.
-type CommitSig Vote
+// BlockIDFlag indicates which BlockID the signature is for.
+type BlockIDFlag byte
 
-// String returns the underlying Vote.String()
-func (cs *CommitSig) String() string {
-	return cs.toVote().String()
+const (
+	// BlockIDFlagAbsent - vote is not included in the Commit.Precommits.
+	BlockIDFlagAbsent BlockIDFlag = iota + 1
+	// BlockIDFlagCommit - voted for the Commit.BlockID.
+	BlockIDFlagCommit
+	// BlockIDFlagNil - voted for nil.
+	BlockIDFlagNil
+)
+
+// CommitSig is a part of the Vote included in a Commit.
+type CommitSig struct {
+	BlockIDFlag      BlockIDFlag `json:"block_id_flag"`
+	ValidatorAddress Address     `json:"validator_address"`
+	Timestamp        time.Time   `json:"timestamp"`
+	Signature        []byte      `json:"signature"`
+}
+
+// NewCommitSigCommit returns new CommitSig with BlockIDFlagCommit.
+func NewCommitSigCommit(signature []byte, valAddr Address, ts time.Time) *CommitSig {
```

Comment:
I know this is just inheriting the name from `NewCommit`, which I don't think is introduced in this PR, but the word "New" in these types makes me wonder what we'll name the next type if we decide to update it again. NewNew?

Reply:
I did not understand, sorry.

But I just thought of a better name - `NewCommitSigForBlock` (will add two more funcs `NewCommitSigForNil` and `NewCommitSigAbsent` later)

Reply:
Ha, sorry--it's a good thing it didn't make sense, because I actually misunderstood something. Sorry about that and please ignore!